### PR TITLE
ENG-11272 multiple small tickets in one

### PR DESCRIPTION
**Pull request by akhanzode**
ENG-11250 - zk lib generilization
ENG-11251 - Pick all consumer properties from file.

### CODE REVIEW

File: src/frontend/org/voltdb/importclient/kafka/KafkaImporterCommitPolicy.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ */
+package org.voltdb.importclient.kafka;
+
+/**
+ * KafkaImporterCommitPolicy determines how offsets are committed to kafka.
+ * @author akhanzode
+ */
+public enum KafkaImporterCommitPolicy {
+
+        COMMIT_POLICY_TIME,
```
Comment (vtkstef) [inline src/frontend/org/voltdb/importclient/kafka/KafkaImporterCommitPolicy.java]: `KafkaImporterCommitPolicy.COMMIT_POLICY_TIME` repeats the notion of commit policy twice. Perhaps `KafkaImporterCommitPolicy.TIME`, and `KafkaImporterCommitPolicy.NONE` would be good enough to describe commit policies

File: src/frontend/org/voltdb/importclient/kafka/KafkaImporterCommitPolicy.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        COMMIT_POLICY_NONE;
+
+        public static final KafkaImporterCommitPolicy fromString(String policy) {
```
Comment (vtkstef) [inline src/frontend/org/voltdb/importclient/kafka/KafkaImporterCommitPolicy.java]: enums already have valueOf(String) already implemented. Why do another?

File: src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporterConfig.java [hunk 3]
```diff
@@ -79,6 +81,19 @@ private KafkaStreamImporterConfig(URI uri, List<HostAndPort> brokers, String top
         m_fetchSize = fetchSize;
         m_soTimeout = soTimeout;
         m_procedure = procedure;
+        KafkaImporterCommitPolicy cp;
+        long triggerValue;
+        try {
+            cp = KafkaImporterCommitPolicy.fromString(commitPolicy);
+            triggerValue = KafkaImporterCommitPolicy.fromStringTriggerValue(commitPolicy, cp);
+        } catch (Exception ex) {
```
Comment (vtkstef) [inline src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporterConfig.java]: can we catch something a little more descriptive like `IllegalArgumentException`

File: src/frontend/org/voltdb/importclient/kafka/KafkaImporterCommitPolicy.java (hunk 2 above)
Comment (akhanzode) [inline src/frontend/org/voltdb/importclient/kafka/KafkaImporterCommitPolicy.java]: because the value is like 1000ms and not enum value from deployment.

**Pull request merged**